### DISCUSSION TITLE
"fetch" command

### DISCUSSION THREAD

**Discussion by ashvinr03** [excalith/excalith-start-page#38] [Ideas]
a very simple way I have implemented this. could be easily customizable with your config json as well.
![Screenshot 2023-05-08 012322](https://user-images.githubusercontent.com/26064522/236740811-679d71a1-38e8-40d5-a7d5-51c2e93f352d.png)

**Comment by excalith** [comment]
Hey @shvin, 

You have a nice addition there, I really like it! I was thinking about making another command for the whole weather thing, including fetching current, hourly and upcoming. However, this would be a nice addition to fetch command as well. 

The way I proceed would be like;
- Make a data structure for weather data to talk through the components and probably cache them into local storage until new data retrieved
- Make a system that allows you to implement your own API request, so you could use any service you want and create the data above
- Fetch will only show current conditions whereas `weather` command will show current, daily and upcoming data

What do you think about this?

**Reply by ashvinr03** [reply]
Very nice ideas, some other ideas I have and would be willing to help with (if you need it) are:

- Having a structure that holds the ASCII art for browsers, or OS' where it would be automatically set but also customizable for the picture portion of the fetch.
- "Sleep Mode" where if the user is inactive for more than a couple minutes but is still on the tab, show the fetch screen until user returns and input is registered again.

**Reply by LevYas** [reply]
I like the weather idea! Probably on another card/command, so a user can select a widget. I love forecasts from Windy, they have nice widgets in i-frames https://www.windy.com/-Embed-widget-on-page/widgets?36.871,30.523,5
